pkg/util/loop/loop.go

```diff
@@ -190,9 +202,16 @@ func (loop *Device) AttachFromFile(image *os.File, mode int, number *int) error
 			syscall.Syscall(syscall.SYS_IOCTL, uintptr(loopFd), CmdClrFd, 0)
```

Comment:
I believe the CmdClrFd syscall can be moved up to L192 above the check for EAGAIN/EBUSY so that it doesn't need to be repeated - once this block is entered we know we have some sort of error and will either be continuing or returning and since we are no longer retrying in here, the file descriptor can be cleared to release the loop device right away.